languagetool-language-modules/pt/src/main/resources/org/languagetool/rules/pt/grammar.xml

```diff
@@ -44227,25 +44227,335 @@ Foundation, Inc., 51 Franklin St, Fifth Floor, Boston, MA 02110-1301, USA.
+        <example correction="abraços">Grato, <marker>abs</marker>.</example>
+	    <example correction="Abraços"><marker>Abs</marker>.</example>
+     </rule>
+  	 <rule>
+        <pattern>
+          <token regexp='yes'>tb|tbm</token>
+        </pattern>
+        <message>Linguagem de Internet. Considere escrever <suggestion>também</suggestion> por extenso.</message>
+        <example correction="também">Eu <marker>tbm</marker>.</example>
+	    <example correction="também">O departamento financeiro <marker>tb</marker> vai participar.</example>
+     </rule>
+     <rule>
+	  <antipattern>
+	  	  <token regexp='yes'>o|um|algum|nenhum|esse|desse|este|deste|neste|nesse|aquele|daquele|do</token>
+          <token>pq</token>
+          <token regexp='yes'>[\.\?!;]</token>
+		  <example>Quero saber o pq.</example>
+		  <example>Isso não tem algum pq.</example>
+		  <example>Isso tem algum pq?</example>
+      </antipattern>
+      <pattern>
+          <token>pq</token>
+          <token regexp='yes'>[\.\?!;]</token>
+      </pattern>
+         <message>Linguagem de Internet. "Por quê" é utilizado no final de frases. Considere escrever <suggestion>por quê</suggestion> por extenso.</message>
+	     <example correction="por quê">Quero saber <marker>pq.</marker></example>
+	     <example correction="por quê">E tudo isso <marker>pq?</marker></example>
+	     <example correction="Por quê">Eu não posso ir. <marker>Pq?</marker></example>
+      </rule>
+      <rule>
+      <pattern>
+         <token regexp='yes'>o|os|um|uns|algum|alguns|nenhum|nenhuns|esse|esses|desse|desses|este|estes|deste|destes|neste|nestes|nesse|nesses|aquele|aqueles|daquele|daqueles|do|dos</token>
+         <marker><token regexp='yes'>pq|pqs</token></marker>
+          <token regexp='yes'>d(?:a|e|o|isso|este|esta|ess[ea]|isto|aquilo|aquela|aquele)s?|n(?:isso|este|esta|ess[ea]|isto|aquilo|aquela|aquele)s?|[\.\?!;]</token>
+      </pattern>
+      <message>Linguagem de Internet; escreva por extenso. A forma <suggestion>porquê</suggestion> é quase sempre precedida por um artigo, um pronome ou numeral, e tem a função de um substantivo.</message>
+      <example correction="porquê">Tem algum <marker>pq</marker> nisso tudo?</example>
+      </rule>
+      <rule>
+        <pattern>
+          <marker><token>pq</token></marker>
+          <token regexp='yes'>raz(?:ão|ões)|motivos?|causas?</token>
+        </pattern>
+		  <message>Linguagem de Internet; escreva por extenso. Sempre que for possível substituir por expressões equivalentes a "(razão/motivo) pelo qual", deve utilizar <suggestion>por que</suggestion>.</message>
+		  <example correction="por que">Não sei <marker>pq</marker> motivo o plano não funcionou.</example>
+		  <example correction="por que">Gostaria de saber <marker>pq</marker> razões recusou o convite.</example>
+	  </rule>
+	  <rule>
+        <pattern>
+		  <token regexp='yes'>o|as|os|a|,</token>
+		  <token regexp='yes'>raz(?:ão|ões)|motivos?|causas?</token>
```

Comment:
maybe it is intended but the "?" will give you either one "(" oder none at all. But the code does not look like you want to include a "("

Reply:
Nope, I will write the words in full then just to be sure

Reply:
I see I forgot to fix one of those, will commit again